src/config.c

```diff
@@ -435,6 +437,47 @@ gboolean parse_config (GromitData *data)
 
           token = g_scanner_get_next_token(scanner);
 
+          if (token != ';')
+            {
+              g_printerr ("Expected \";\"\n");
+              goto cleanup;
+            }
+        }
+      else if (token == G_TOKEN_SYMBOL && scanner->value.v_symbol == HISTORY_SYMBOL_VALUE)
+        {
+          /*
+           * History steps definition
+           */
+          gpointer key_type = scanner->value.v_symbol;
+          token = g_scanner_get_next_token(scanner);
+
+          if (token != G_TOKEN_EQUAL_SIGN)
+          {
+            g_scanner_unexp_token (scanner, G_TOKEN_EQUAL_SIGN, NULL,
+                                   NULL, NULL, "aborting", TRUE);
+            goto cleanup;
+          }
+
+          token = g_scanner_get_next_token (scanner);
+
+          if (token != G_TOKEN_FLOAT)
+          {
+            g_scanner_unexp_token (scanner, G_TOKEN_FLOAT, NULL,
+                                   NULL, NULL, "aborting", TRUE);
+            goto cleanup;
+          }
+
+          data->undo_steps = (gint) scanner->value.v_float;
+          data->undobuffer = calloc(data->undo_steps, sizeof(cairo_surface_t*));
```

Comment:
This should not happen in config.c, rather in the place where the undo buffer is alloced.